Add a feature to view comments

doko-desuka opened the pull request:
This adds menu "Comments" to the "More..." dialog from the context menu. It takes you to a directory showing the "top level" comments made on the video, with the comment text as the plot in each item.  
The items are folders, entering a top-level comment will show the replies inside.

![ytComments0](https://user-images.githubusercontent.com/31810999/79536036-6e571e00-8055-11ea-99e7-d4725afb6419.png)

![ytComments](https://user-images.githubusercontent.com/31810999/79535563-36031000-8054-11ea-84ae-1a6c31e61521.png)

doko-desuka commented:
One thing I noticed is that the automatic pagination from the add-on is not kicking in. If I enter a comment with 457 replies it only shows 50, with no "next page" item.

mrmvergeer commented:
I think that's an API restriction:  only the last 50 comments

anxdpanic commented on resources/lib/youtube_plugin/youtube/helper/utils.py:
```diff
@@ -43,6 +43,27 @@ def extract_urls(text):
+def make_comment_item(snippet, uri, total_replies=0):
+    label = '[B]{}[/B]'.format(kodion.utils.to_utf8(snippet['authorDisplayName']))
+    if (snippet['publishedAt'] != snippet['updatedAt']):
+        label += ' (edited)'
```
`edited` needs to be localized

doko-desuka replied:
Of course, I completely forgot about localization.

anxdpanic commented on resources/lib/youtube_plugin/youtube/helper/utils.py:
```diff
@@ -43,6 +43,27 @@ def extract_urls(text):
+    
+    extra_info = [ ]
+    if snippet['likeCount']:
+        extra_info.append('Likes: {likes:d}'.format(likes=snippet['likeCount']))
```
`Likes` needs to be localized

anxdpanic commented on resources/lib/youtube_plugin/youtube/helper/utils.py:
```diff
@@ -43,6 +43,27 @@ def extract_urls(text):
+    if snippet['likeCount']:
+        extra_info.append('Likes: {likes:d}'.format(likes=snippet['likeCount']))
+    if total_replies:
+        extra_info.append('Replies: {total_replies:d}'.format(total_replies=total_replies))
```
`Replies` needs to be localized

doko-desuka commented:
@anxdpanic I added the localizations as you requested, if you think of anything else please let me know.  
I also changed some visual things to make it easier to use:  

![ytComments2](https://user-images.githubusercontent.com/31810999/79630851-37ebd280-812b-11ea-8548-d8edeb9a91b8.png)

The comment texts are now also added to the item labels to make some of them quicker to read when they're only a single line. It also compacts the "likes" and "replies" values into those colored labels.

@mrmvergeer thanks for looking into it. I noticed that for some reason I was setting `process_next_page=False`, which was causing this, so now it's fixed.  
But it's not that great anyway because the YouTube API only serves comments in reverse order: newest first, oldest last (this is different than what the website does, it shows oldest replies first, newest last). I filed a bug report, I hope they change this in the future and add the same sorting as the website uses.

doko-desuka commented:
Minor change, removed the quotes from the text (it was just adding noise).

![ytComments3](https://user-images.githubusercontent.com/31810999/79632270-b1d48980-8134-11ea-99d8-d12161014cb0.png)

anxdpanic commented:
@doko-desuka Thanks again!, it all looks good to me now.

anxdpanic commented:
added to changelog in #789

doko-desuka commented:
@anxdpanic np, thanks for the quick review

anxdpanic commented:
Included in beta1, https://github.com/jdf76/plugin.video.youtube/releases/tag/6.7.1-dev
